### PR TITLE
Create Ackermann function (dart)

### LINKED ISSUE #5407: Ackermann Function in dart
## 💥 Proposal

Ackermann Function in dart

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

### PR DESCRIPTION
**Pull request by ashuvssut**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

@HarshCasper I have Created Ackermann function algo in Dart

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

closes #5407

### CODE REVIEW

File: Dart/maths/Ackermann_function.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import 'dart:io';
```
Comment (HarshCasper) [inline Dart/maths/Ackermann_function.dart:9]: Add a beginner-friendly description

File: Dart/maths/Ackermann_function.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  print("Input first Number");
+  int m = int.parse(stdin.readLineSync());
+  print("Input second Number");
+  int n = int.parse(stdin.readLineSync());
+  print(Ackermann_Function(m, n));
+}
+
+/*
+time complexity:-O(mA(m, n)) to compute A(m, n)
+space complexity:- O(m) to compute A(m, n)
```
Comment (HarshCasper) [inline Dart/maths/Ackermann_function.dart]: What do these `A(m, n)` mean? Can you please add a note?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Why code looks similar to gfg code,

**Comment by ashuvssut** [comment]
@ankitaggarwal23 

 in ackermann function A(m,n),
    * If m=0, then A(m,n) = n+1 
    * If m != 0 but n=0, then A(m,n)=A(m-1,1)
    * If m and n both are not zero then... A(m,n) = A( m-1, A(m,n-1) )
    
upon these 3 conditions of an Ackermann function anyone would write the same code only. it's just recursions that follows those three conditions of Ackermann function. 

and about the time complexity lines... yes, I copied the two lines of the gfg article. I didn't find any direct answer in Stack overflow and Stack exchange.


@ankitaggarwal23 see line 17 and please help n tell what must be the time complexity

`Ackermann_Function(m - 1, Ackermann_Function(m, n - 1));`

I will update that time complexity line as per your suggestions

File: Dart/maths/Ackermann_function.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
```
Comment (ankitaggarwal23) [inline Dart/maths/Ackermann_function.dart]: Don't be lazy, Capitalize the first letter T and S, and what is mA?

File: Dart/maths/Ackermann_function.dart [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+Examples:-
+m=1,n=2
+output=4
+m=3,n=3
+output=61
+*/
```
Comment (ankitaggarwal23) [inline Dart/maths/Ackermann_function.dart]: Write the output in the way your code is producing.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Remove the `.gitignore` file

**Comment by ashuvssut** [comment]
https://replit.com/@AshutoshKhandua/LimegreenSuperMosaic#main.dart

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**